### DISCUSSION TITLE
RFC: Convert optional dependencies to use "extras", to lower the dependencies footprint and ease installation of binary extensions by avoiding them if not necessary

### DISCUSSION THREAD

**Discussion by Lenormju** [msqd/harp#601] [Requests for comments]
The dependency `asyncmy` seems to be troublesome to install, lacking pre-built wheels for many platforms. This means that in order to install it, it is sometimes required to have the corresponding `python3-dev` installed on Unix, or `Microsoft Visual C++ 14.0 or greater` on Windows.

We discussed about moving `asyncmy` to "extras", but it will have an impact on the project (installation, build, CI, distribution, ...). Is it a good idea ? We should ponder the consequences.

**Comment by hartym** [comment]
We probably should implement extra for mysql, postgresql, probably notifications like slack, etc. and all optional dependencies candidates. The direct issue with this is that it requires a matrix of builds that may start to create a very long list of image builds (for docker installations). But this is the same problem as with custom builds for custom extensions. Needs to be well thought before going there.
